packages/block-editor/src/components/inserter/block-types-tab.js

```diff
@@ -30,6 +30,17 @@ const MAX_SUGGESTED_ITEMS = 6;
  */
 const EMPTY_ARRAY = [];
 
+// Blocks that should be prioritized in the Inserter tab.
+const PRIORITIZED_BLOCKS = {
```

Comment:
Instead of keeping this list here why not just move the imports properly (or reorder the actual array used when registering the blocks)?

Reply:
Reordering the blocks wouldn't work well, as we want to prioritize block variations too and we might want some block's variations not to appear sequentially (ex. `Header`, `Posts List`, `Footer`.

The other place this sorting could happen would be in [the selector](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/selectors.js#L1682) which is used in other places as well and didn't seem to me the best place to put extra logic where is not needed. I think the only place this is needed is the `block types tab` which is also the only place we split them by category.

Reply:
For 5.9, can we just reorder the import statements? This other system seems too complex for what we are trying to achieve.

Reply:
We don't want to prioritize variations over their source block. The way to do so would be to hide the source from the inserter, in any case (Query should only appear when you search for it, as it's an advance block; users would insert "Post list" or "Page list" or "Product list" etc).

Reply:
Let's avoid all custom handling on this layer, since it's not a matter of order but a matter of when to show / hide a source block.

Reply:
We don't support right now controlling the availability of blocks per inserter (main, slash, etc..). I think the alternative for this would be to change the order of items to show its block's variations right after the source block and probably make `Posts List` the default variation for `Query Loop`. That would mean though that if someone types `Query` the `Posts Lists` would only be shown. 

What do you think about this?

Reply:
Let's do the order first and look at everything else later